biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/ReducedChemCompProvider.java

```diff
@@ -46,54 +46,46 @@
 	public ReducedChemCompProvider(){
 		logger.debug("Initialising ReducedChemCompProvider");
 	}
-	
-	
-	public ChemComp getEmptyChemComp(){
-		ChemComp comp = new ChemComp();
-		
-		comp.setOne_letter_code("?");
-		comp.setPolymerType(PolymerType.unknown);
-		comp.setResidueType(ResidueType.atomn);
-		return comp;
-	}
-	
+
+
 	@Override
 	public ChemComp getChemComp(String recordName) {
 		String name = recordName.toUpperCase().trim();
-		InputStream inStream = this.getClass().getResourceAsStream("/chemcomp/"+name + ".cif.gz");
-
-		if ( inStream == null){
-			//System.out.println("Could not find chem comp: " + name + " ... using generic Chem Comp");
-			// could not find the chem comp definition for this in the jar file
-			logger.debug("Getting empty chem comp for {}",name);
-			ChemComp cc = getEmptyChemComp();
-			cc.setId(name);
-			return cc;
-		}
+		try {
+			try(InputStream inStream = this.getClass().getResourceAsStream("/chemcomp/"+name + ".cif.gz")) {
```

Comment:
Shouldn't the try-with be inside the first try? The second try is redundant, isn't it?